src/goto-instrument/code_contracts.cpp

```diff
@@ -242,6 +259,70 @@ void code_contractst::code_contracts(
       apply_contract(goto_function.body, it);
 }
 
+void code_contractst::assert_ensures(
+  goto_programt &goto_program,
+  goto_programt::targett target)
+{
+  const code_function_callt &call = target->get_function_call();
+
+  // we don't handle function pointers
+  if(call.function().id() != ID_symbol)
+    return;
+
+  const irep_idt &function = to_symbol_expr(call.function()).get_identifier();
+  const symbolt &f_sym = ns.lookup(function);
+  const code_typet &type = to_code_type(f_sym.type);
+
+  exprt ensures = static_cast<const exprt &>(type.find(ID_C_spec_ensures));
+
+  // is there a contract?
+  if(ensures.is_nil())
+    return;
+
+  // replace formal parameters by arguments, replace return
+  replace_symbolt replace;
+
+  // TODO: return value could be nil
+  if(type.return_type() != empty_typet())
+  {
+    symbol_exprt ret_val(CPROVER_PREFIX "return_value", call.lhs().type());
+    replace.insert(ret_val, call.lhs());
+  }
+
+  // formal parameters
+  code_function_callt::argumentst::const_iterator a_it =
+    call.arguments().begin();
+  for(code_typet::parameterst::const_iterator p_it = type.parameters().begin();
+      p_it != type.parameters().end() && a_it != call.arguments().end();
+      ++p_it, ++a_it)
```

Comment:
Minor style comment - I'd put explicit braces in for the body of the for loop here, even though it's technically only "one statement" in the loop.... but given its' multiple lines of text I think it would be more readable with explicit braces.

Reply:
Thanks, that makes sense :)